src/behavior.ts

```diff
@@ -24,147 +24,257 @@
+            event.preventDefault();
+            event.stopPropagation();
+
+            this.selectionManager.showContextMenu(dataPoint.data.identity, {
+                x: event.clientX,
+                y: event.clientY
+            });
+        });
+
+        const handleEmptyContextMenu = (event: MouseEvent) => {
+            event.preventDefault();
+            event.stopPropagation();
+
+            const emptySelection = {
+                "measures": [],
+                "dataMap": {
+                }
+            };
+
+            this.selectionManager.showContextMenu(emptySelection, {
+                x: event.clientX,
+                y: event.clientY
+            });
 
-        const emptySelection = {
-            "measures": [],
-            "dataMap": {
-            }
         };
 
-        clearCatcher.on("contextmenu", () => {
-            const event: MouseEvent = <MouseEvent>getEvent() || <MouseEvent>window.event;
-            if (event) {
-                selectionHandler.handleContextMenu(
-                    <BaseDataPoint>{
-                        identity: emptySelection,
-                        selected: false
-                    },
-                    {
-                        x: event.clientX,
-                        y: event.clientY
-                    });
+        this.options.centerLabel.on("contextmenu", handleEmptyContextMenu);
+        this.options.lineLabels.on("contextmenu", handleEmptyContextMenu);
+        this.options.clearCatcher.on("contextmenu", handleEmptyContextMenu);
+    }
+
+    private bindKeyboardEvents(): void {
+        this.options.selection.on("keydown", (event: KeyboardEvent, d: d3PieArcDatum<AsterDataPoint>) => {
+            if (event.code == EnterCode || event.code == SpaceCode) {
                 event.preventDefault();
-                event.stopPropagation();
+                this.selectDataPoint(d.data, event.ctrlKey || event.metaKey || event.shiftKey);
+                this.onSelectCallback();
             }
         });
     }
 
-    public renderSelection(hasSelection: boolean) {
-        this.selection.attr("aria-selected", (d) => {
+    private selectDataPoint(dataPoint: AsterDataPoint | LegendDataPoint, multiSelect: boolean = false): void {
+        if (!dataPoint || !dataPoint.identity) return;        
+
+        const selectedIds: ISelectionId[] = <ISelectionId[]>this.selectionManager.getSelectionIds();
+        const isSelected: boolean = this.isDataPointSelected(dataPoint, selectedIds);
+
+        const selectionIdsToSelect: ISelectionId[] = [];
+        if (!isSelected) {
+            dataPoint.selected = true;
+            selectionIdsToSelect.push(dataPoint.identity);
+        } else {
+            // toggle selected back to false
+            dataPoint.selected = false;
+            if (multiSelect) {
+                selectionIdsToSelect.push(dataPoint.identity);
+            }
+        }
```

Comment:
This one can be replaced with:
``` ts
dataPoint.selected = !isSelected
if(!isSelected || multiSelect) {
   selectionIdsToSelect.push(dataPoint.identity);
}
```
Also not sure that we should set to `false` `.selected` if `multiSelect == true`. Sounds strange that we push it to selected but marks as unselected

Reply:
I remember that it was my attempt to allow users to toggle the data point by clicking on it. Settings `dataPoint.selected = false;` doesn't do anything, because the array `selectionIdsToSelect` is empty (if no multi-selection is applied). After the clicking the callback `onSelectCallback` is run, which reads the array and updates `selected` property.

@AleksSavelev 
Anyway, I applied your suggested code and it looks better. Though, I want to mention that there's a change in selectionManager. 

Here's my new code:
```TypeScript
if (selectionIdsToSelect.length === 0) {
    this.selectionManager.clear();
} else {
    this.selectionManager.select(selectionIdsToSelect, multiSelect);
}
```

I remember that you could just pass empty array to `selectionManager.select()` and it's smart enough to clear selections if the array is empty. But in the recent versions of `ISelectionManager` it doesn't do it. Therefore I had to update the code as it shown before.